app/src/androidTest/java/ch/epfl/sdp/appart/MapUITest.java

```diff
@@ -127,4 +131,68 @@ public void markerTest() throws InterruptedException {
                         " CLICKED")), 10000) | device.hasObject(By.descContains(card.getCity() + "CLICKED"));
         assertThat(isMarkerClicked, is(true));
     }
+
+    @Before
+    public void init() {
+        Intents.init();
+    }
+
+    @Test
+    public void markerClickTest() throws InterruptedException {
+
+
+        UiDevice device =
+                UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+
+        boolean foundMap = device.wait(Until.hasObject(By.desc("MAP READY")),
+                10000);
+        assertThat(foundMap, is(true));
+
+        Thread.sleep(2000);
+
+        List<Card> cards = databaseService.getCards().join();
+
+        Card card = cards.get(0);
+
+        Location loc =
+                locationService.getLocationFromName(card.getCity()).join();
+        mapService.centerOnLocation(loc, true);
+
+
+        boolean isMarkerPresent =
+                device.hasObject(By.descContains(card.getCity())) | device.wait(Until.hasObject(By.descContains(card.getCity())), 10000);
+        assertThat(isMarkerPresent, is(true));
+
+
+        UiObject2 marker = device.findObject(By.descContains(card.getCity()));
+        marker.click();
+
+        Thread.sleep(2000);
+
+        //https://stackoverflow.com/questions/42505274/android-testing-google
+        // -map-info-window-click
+        Display display =
+                InstrumentationRegistry.getInstrumentation().getTargetContext().getDisplay();
+        Point size = new Point();
+        display.getRealSize(size);
+        int screenWidth = size.x;
+        int screenHeight = size.y;
+        int x = screenWidth / 2;
+        int y = (int) (screenHeight * 0.35);
+
+        device.click(x, y);
```

Comment:
are these values picked to make sure cirrus clicks at the right spot? should this work for any emulator?

Reply:
This comes from a stackoverflow post. And in fact, when you click on a marker the info window takes approximately the whole upper half of the screen. The constant used in the post was 0.43 but it doesn't seem to work here, that's why I lowered it a little. However, since this is a proportion and since we programmatically get the screen size of the actual device I don't think this will be an issue on any device.